src/v/cluster/scheduling/constraints.h

```diff
@@ -27,4 +29,8 @@ distinct_from(const std::vector<model::broker_shard>&);
 
 soft_constraint_evaluator least_allocated();
 
+soft_constraint_evaluator distinct_rack(
+  const std::vector<model::broker_shard>&,
+  ss::noncopyable_function<ss::sstring(model::node_id)>);
```

Comment:
do we need to pass in a function here ? i think that there is only one implementation that doesn't have to be customizable